src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -982,14 +980,7 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof SSLCallback) {
                     SSLCallback sslCallback = (SSLCallback) callback;
-
-                    try {
-                        X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
-                        verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
```

Comment:
It's not strictly necessary to limit this callback to EXTERNAL. Calling verifyEvidence allows authorization to occur at this point but it also still allows additional identity changes without an authorization check.